dashboard/modules/reporter/reporter_head.py

```diff
@@ -112,6 +142,243 @@ async def get_cluster_status(self, req):
                 cluster_status=debug_status(formatted_status_string, error),
             )
 
+    async def get_task_ids_running_in_a_worker(self, worker_id: str) -> List[str]:
+        """
+        Retrieves the task IDs of running tasks associated with a specific worker.
+
+        Args:
+            worker_id (str): The ID of the worker.
+
+        Returns:
+            List[str]: A list containing the task IDs of all the running tasks associated with the worker.
+        """
+
+        option = ListApiOptions(
+            filters=[("worker_id", "=", worker_id), ("state", "=", "RUNNING")],
+            detail=True,
+            timeout=10,
+        )
+        ## Call the state API to get all tasks in a worker
+        tasks_in_a_worker_result = await self._state_api.list_tasks(option=option)
+        tasks_in_a_worker = tasks_in_a_worker_result.result
+
+        ## Get task_id from each task in a worker
+        task_ids_in_a_worker = []
+        if tasks_in_a_worker is not None:
+            task_ids_in_a_worker = [
+                task.get("task_id")
+                for task in tasks_in_a_worker
+                if task and "task_id" in task
+            ]
+        return task_ids_in_a_worker
+
+    def set_up_state_api(self):
+        """
+        Set up the state API in order to fetch task information.
+        """
+        gcs_channel = self._dashboard_head.aiogrpc_gcs_channel
+        self._state_api_data_source_client = StateDataSourceClient(
+            gcs_channel, self._dashboard_head.gcs_aio_client
+        )
+        self._state_api = StateAPIManager(self._state_api_data_source_client)
```

Comment:
I wonder if this needs to be a separate function. Might be simpler to just inline this code in the `__init__`

Reply:
I agree, a little bit over-engineering here